src/numbuf/python/src/pynumbuf/numbuf.cc

```diff
@@ -275,12 +345,9 @@ static PyObject* store_list(PyObject* self, PyObject* args) {
   }
   CHECK(error_code == PlasmaError_OK);
 
-  auto target = std::make_shared<FixedBufferStream>(LENGTH_PREFIX_SIZE + data, size);
-  std::shared_ptr<arrow::ipc::FileWriter> writer;
-  ipc::FileWriter::Open(target.get(), batch->schema(), &writer);
-  writer->WriteRecordBatch(*batch, true);
-  writer->Close();
-  *((int64_t*)data) = size;
+  auto target = std::make_shared<FixedBufferStream>(LENGTH_PREFIX_SIZE + data, total_size);
```

Comment:
Yes, my thinking was that you could contribute the multithreaded memcpy to Arrow and we could add an option to this class to switch between that and std::memcpy. Let me know if I can help